### PR TITLE
Update to support Plugins v2

### PR DESCRIPTION
**Pull request by lieut-data**
Currently point at plugins-2 branch, won't merge until that branch hits `master` at least. Later we can update to version 5.2.0.

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -12,25 +12,15 @@ import (
...
-		return err
-	}
+	plugin.MattermostPlugin
 
-	return nil
+	links atomic.Value
 }
 
 // OnConfigurationChange is invoked when configuration changes may have been made.
 func (p *Plugin) OnConfigurationChange() error {
```
[the remark this answers is not in the document]
Comment (crspeller) [reply server/plugin.go]: No objections.

File: server/main.go [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package main
 
 import (
-	"github.com/mattermost/mattermost-server/plugin/rpcplugin"
+	"github.com/mattermost/mattermost-server/plugin"
 )
 
 func main() {
-	rpcplugin.Main(&Plugin{})
+	plugin.ClientMain(&Plugin{})
```
[the remark this answers is not in the document]
Comment (crspeller) [reply server/main.go]: No I renamed it because it made sense with the rest of the client* naming convention.

**Comment by lieut-data** [comment]
@jasonblais, that seems to be the strategy we're using for https://github.com/mattermost/mattermost-plugin-zoom/pull/5. Any reason not to merge to `master`?

(My specific comment was with respect to the `Gopkg.toml` dependency pointing at the plugins-2 branch of mattermost-server, which ought to be pointed at `master` once `plugins-2` is merged to that repository.)

**Comment by jasonblais** [comment]
Sorry, I was confused here. For some reason I thought this was the server repo..all good 😓

**Comment by lieut-data** [comment]
I've pointed `Gopkg.toml` at `master` and updated the dependencies.

**Pull request merged**